specification/networkanalytics/resource-manager/Microsoft.NetworkAnalytics/stable/2023-11-15/networkanalytics.json

```diff
@@ -0,0 +1,2954 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "dataProductName",
+            "in": "path",
+            "description": "The data product resource name",
+            "required": true,
+            "type": "string",
+            "minLength": 3,
+            "maxLength": 63,
+            "pattern": "^[a-z][a-z0-9]*$"
+          },
+          {
+            "name": "dataTypeName",
+            "in": "path",
+            "description": "The data type name.",
+            "required": true,
+            "type": "string",
+            "minLength": 3,
+            "maxLength": 63,
+            "pattern": "^[a-z][a-z0-9-]*$"
+          },
+          {
+            "name": "properties",
+            "in": "body",
+            "description": "The resource properties to be updated.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DataTypeUpdate"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "ARM operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/DataType"
+            }
+          },
+          "202": {
+            "description": "Resource update request accepted.",
+            "headers": {
+              "Retry-After": {
+                "type": "integer",
+                "format": "int32",
+                "description": "The Retry-After header can indicate how long the client should wait before polling the operation status."
+              },
+              "Location": {
+                "type": "string",
+                "description": "The Location header contains the URL where the status of the long running operation can be checked."
+              }
+            }
+          },
+          "default": {
+            "description": "An unexpected error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DataTypes_Update_MaximumSet_Gen": {
+            "$ref": "./examples/DataTypes_Update_MaximumSet_Gen.json"
+          },
+          "DataTypes_Update_MaximumSet_Gen - generated by [MinimumSet] rule_MinimumSet_Gen": {
+            "$ref": "./examples/DataTypes_Update_MinimumSet_Gen.json"
+          }
+        },
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.NetworkAnalytics/dataProducts/{dataProductName}/dataTypes/{dataTypeName}/deleteData": {
+      "delete": {
```

Comment:
>delete [](http://example.com/codeflow?start=7&length=6)

This is incorrect. You should be deleting the root level resource itself , i,e dataTypes/{dataTypeName} according to REST semantics. This is getting flagged in your tests as well 

[https://github.com/Azure/azure-rest-api-specs/pull/25577/checks?check_run_id=17094650089](https://github.com/Azure/azure-rest-api-specs/pull/25577/checks?check_run_id=17094650089)

Reply:
@raosuhas 
This is a specific post action API to our service, and it not for delete of any RT. we have our release mid October so completing the review, ASAP will help us.  if required we can have a short call.
Note: All these changes are reviewed in private repo, we just want to copy the stable API to public repo as part of this PR.

Reply:
Hi @raosuhas , we have addressed this.
Now the resource DataType has corresponding delete operation and added a post operation for data deletion as well.